benchmark/infrastructure/BenchmarkConfiguration.h

```diff
@@ -35,117 +142,144 @@ class BenchmarkConfiguration {
 
  public:
   /*
-  @brief If the underlying JSON can be accessed with
-  `data_[firstKey][secondKey][....]` then return the resulting value of this
-  recursive access, interpreted as a given type. Else return an empty
-  `std::optional`.
+  @brief Add the given configuration option in such a way, that it can be
+  accessed by calling `getConfigurationOptionByNestedKeys` with the here given
+  keys, follwed by the name of the configuration option.
+  Example: Given a configuration option `numberOfRows` and the keys
+  `"generalOptions", 1, "Table"`, then it can be accessed with
+  `"generalOptions", 1, "Table", "numberOfRows"`.
+  */
+  template <KeyForJson... Keys>
+  void addConfigurationOption(const BenchmarkConfigurationOption& option,
+                              const Keys&... keys) {
+    // All numeric key, must be `>=0`.
+    AD_CONTRACT_CHECK(allArgumentsBiggerOrEqualToZero(keys...));
+
+    // The position in the json object literal, our keys point to.
+    const nlohmann::json::json_pointer ptr{
+        createJsonPointer(AD_FWD(keys)..., option.getIdentifier())};
 
-  @tparam ReturnType The type the resulting value should be interpreted as.
+    /*
+    The first key must be a string, not a number. Having an array at the
+    highest level, would be bad practice, because setting and reading options,
+    that are just identified with numbers, is rather difficult.
+    */
+    if constexpr (sizeof...(keys) > 0) {
+      constexpr auto firstIsString = []<typename T, typename...>() constexpr {
+        return isString<T>;
+      };
+
+      if (!firstIsString.template operator()<Keys...>()) {
+        throw ad_utility::Exception(
+            absl::StrCat("Key error: The first key in '", ptr.to_string(),
+                         "' isn't a string."));
```

Comment:
You can use `AD_THROW`, then you even get line numbers. 
And maybe the error message needs more context (Tell the user that the error is when adding configurations and what the keys were, and why this is wrong).

Reply:
> You can use AD_THROW, then you even get line numbers.

Alright, funny thing: `AD_THROW` just throws an `ad_utility::Exception` in the same way, I do. I could save on one word (`throw`), but not sure, if that is worth anything.

Reply:
I've tried to improve the error message in [5e1d10f](https://github.com/ad-freiburg/qlever/pull/977/commits/5e1d10fa09b6fccbfea610dda8bd41c6ba9c2ea3).

I've added all requested information, except "what the keys were", because that was already included.